src/index.js

```diff
@@ -22,11 +26,54 @@ const products = [
   { name: 'car', category: 'transportation' },
   { name: 'laptop', category: 'computer' },
 ]
+// function addProducts(){
+//   let productList = document.getElementsByClassName(product-list);
+
+// }
 // Target the ul element in index.html and render the above list
 
+let productList = document.querySelector('.product-list');
+products.map(product => {
+  let li = document.createElement("li");
+  li.innerText = `${product.name}`;
+  productList.appendChild(li);
+})
+
 // 7. Target the button in index.html, attach an event listener to it. Every time we click
 // on the button, it will toggle the product list. Show => hide, hide => show
+let btnToggle = document.querySelector('.toggle');
+
+btnToggle.addEventListener("click", (e) => {
+productList.style.display = productList.style.display == "none" ? "block" : "none";
+});
 
 // 8. Target the input element and attach an event listener to it.
 // Once we click search button, you need to use the text input to search for the relevant product name from the list above
 // and show it in the <span> tag
+
+
+let searchResult = document.querySelector('.result');
+btnSearch = document.querySelector('.search-button');
+input = document.querySelector('.search-input');
+
+
+btnSearch.addEventListener("click", (e) => {
+  e.preventDefault()
+  console.log('button clicked', input.value);
+  query = input.value;
+  console.log(query);
+  result = products.find(product => product.name ==query);
+  console.log(result);
+  searchResult.innerText = result.name;
```

Comment:
Be careful here, `find` will return undefined if no element matches the condition. You need to handle that case